### PR TITLE
[SPARK-35678][ML][FOLLOWUP] Revert changes in ANN

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
revert changes related to ANN

### Why are the changes needed?
using the new `softmax` may cause flaky failure 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
reverted testsuite

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Thanks for the quick fix!

File: python/pyspark/ml/tests/test_algorithms.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def test_raw_and_probability_prediction(self):
         result = model.transform(test).head()
         expected_prediction = 2.0
         expected_probability = [0.0, 0.0, 1.0]
-        expected_rawPrediction = [-11.824, -8.298, 22.5299]
+        expected_rawPrediction = [-11.6081922998, -8.15827998691, 22.17757045]
         self.assertTrue(result.prediction, expected_prediction)
         self.assertTrue(np.allclose(result.probability, expected_probability, atol=1E-4))
         self.assertTrue(np.allclose(result.rawPrediction, expected_rawPrediction, rtol=0.1))
```
Comment (HyukjinKwon) [inline python/pyspark/ml/tests/test_algorithms.py:89]: BTW, @zhengruifeng, if the changes in the main codes are correct, we could just simply increase `rtol` to  something like `0.2` too.

Comment (HyukjinKwon) [reply python/pyspark/ml/tests/test_algorithms.py:89]: I will leave it to you @zhengruifeng

**Review by HyukjinKwon — APPROVED** [review]
LGTM either way

**Comment by HyukjinKwon** [comment]
tests passed. I will merge this in first to get the build passed

File: python/pyspark/ml/tests/test_algorithms.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/ml/tests/test_algorithms.py:89]: 3.2 is coming, let's revert it for now.

Comment (HyukjinKwon) [reply python/pyspark/ml/tests/test_algorithms.py:89]: 👌

**Pull request closed without merging**